components/doc/datatable/rowselection/singledoc.js

```diff
@@ -67,16 +67,16 @@ import { InputSwitch, InputSwitchChangeEvent } from 'primereact/inputswitch';
 import { ProductService } from './service/ProductService';
 
 interface Product {
-    id: string;
-    code: string;
-    name: string;
-    description: string;
-    image: string;
-    price: number;
-    category: string;
-    quantity: number;
-    inventoryStatus: string;
-    rating: number;
+    id?: string;
```

Comment:
Why are these all possibly undefined? Doesn't the response from the data provide that object without any missing fields?

Reply:
I always copy and paste this object around while testing so I made them all optional to make my life easier when testing 😄